R/model.R

```diff
@@ -856,3 +858,114 @@ variational_method <- function(data = NULL,
   CmdStanVB$new(runset)
 }
 CmdStanModel$set("public", name = "variational", value = variational_method)
+
+#' Run Stan's standalone generated quantities method
+#'
+#' @name model-method-generate-quantities
+#' @aliases generate_quantities
+#' @family CmdStanModel methods
+#'
+#' @description The `$generate_quantities()` method of a [`CmdStanModel`] object
+#'   runs Stan's standalone generated quantities to obtain generated quantities
+#'   based on previously fitted parameters.
+#'
+#' @section Usage:
+#'   ```
+#'   $generate_quantities(
+#'     fitted_params,
+#'     data = NULL,
+#'     seed = NULL,
+#'     output_dir = NULL,
+#'     parallel_chains = getOption("mc.cores", 1),
+#'     threads_per_chain = NULL
+#'   )
+#'   ```
+#'
+#' @section Arguments:
+#'   * `fitted_params`: (multiple options) The parameter draws to use. One of the following:
+#'     - A [CmdStanMCMC] fitted model object.
+#'     - A character vector of paths to CmdStan CSV output files containing
+#'     parameter draws.
+#'   * `data`, `seed`, `output_dir`, `parallel_chains`, `threads_per_chain`:
+#'   Same as for the [`$sample()`][model-method-sample] method.
+#'
+#' @section Value: The `$generate_quantities()` method returns a [`CmdStanGQ`] object.
+#'
+#' @template seealso-docs
+#'
+#' @examples
+#' \dontrun{
+#' # first fit a model using MCMC
+#' mcmc_program <- write_stan_tempfile(
+#'   "data {
+#'     int<lower=0> N;
+#'     int<lower=0,upper=1> y[N];
+#'   }
+#'   parameters {
+#'     real<lower=0,upper=1> theta;
+#'   }
+#'   model {
+#'     y ~ bernoulli(theta);
+#'   }"
+#' )
+#' mod_mcmc <- cmdstan_model(mcmc_program)
+#'
+#' data <- list(N = 10, y = c(1,1,0,0,0,1,0,1,0,0))
+#' fit_mcmc <- mod_mcmc$sample(data = data, seed = 123, refresh = 0)
+#'
+#' # stan program for standalone generated quantities
+#' # (could keep model block, but not necessary so removing it)
+#' gq_program <- write_stan_tempfile(
+#'   "data {
+#'     int<lower=0> N;
+#'     int<lower=0,upper=1> y[N];
+#'   }
+#'   parameters {
+#'     real<lower=0,upper=1> theta;
+#'   }
+#'   generated quantities {
+#'     int y_rep[N] = bernoulli_rng(rep_vector(theta, N));
+#'   }"
+#' )
+#'
+#' mod_gq <- cmdstan_model(gq_program)
+#' fit_gq <- mod_gq$generate_quantities(fit_mcmc, data = data, seed = 123)
+#' str(fit_gq$draws())
+#'
+#' library(posterior)
+#' as_draws_df(fit_gq$draws())
+#' }
```

Comment:
@rok-cesnovar Added this example using the new `write_stan_tempfile()`. What do you think?

Reply:
Seems like a good idea for showcasing stuff yeah. I am in favor.